### ISSUE TITLE
Would like support for new CMake based JUCE

### ISSUE THREAD

**Issue by christofmuc** [FigBug/Gin#16]
Hi! Love your work! I integrate the Gin module into a bigger application using the new JUCE 6's CMake mechanism. 

I wrote a mini CMakeLists.txt for the Gin library, which works as expected on Linux and Windows, but utterly fails on the Mac build. Me, not being Mac savvy, have no idea what is going wrong. It seems the #import pragma somehow is wrong. 

The CMakeLists.txt I have written is here: https://github.com/christofmuc/KnobKraft-orm/blob/master/third_party/CMakeLists.txt

I include Gin as a git submodule. 

The error message on the Mac is in the build log (hope the access permissions allow you to follow that link:
https://dev.azure.com/christof0759/KnobKraft/_build/results?buildId=292&view=logs&j=e0fca624-2401-5624-f540-1c188feac080&t=95b3b2e9-1390-5b19-59f2-dbccb23e7f0a

Basically it reads:

    In file included from /Users/runner/work/1/s/third_party/Gin/modules/gin/gin.cpp:30:
    In file included from /Applications/Xcode_11.3.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer  /SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:8:
    /Applications/Xcode_11.3.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System /Library/Frameworks/Foundation.framework/Headers/NSObjCRuntime.h:492:1: error: expected unqualified-id
    @class NSString, Protocol;
    ^

The error seems to indicate it has interpreted the #import as Objective-C?

Sorry if that is confusing, but maybe it's an opportunity for you to update Gin to the latest JUCE?

**Comment by FigBug** [comment]
I'm not really sure how CMake works or what I need to change on my end, but it looks like it's trying to build the gin.cpp instead of the gin.mm

**Comment by christofmuc** [comment]
Thank you! Funny, I didn't even know what an mm file is and it seems super-counterintuitive to compile C++ wrapped in an mm file which is Objective-C, but if I change the line to mm it works. Obviously I know nothing about Mac compilation ;-)

CMake is pretty nice in building more complex software packages which integrate lots of other third party C++ libraries, and JUCE 6 now has native support for it. Check it out! My Open Source projects KnobKraft and Jammernetz both use it extensively, and have lots of CMake code to serve as an example.
